### DISCUSSION TITLE
Accessing and displaying the Field description attribute

### DISCUSSION THREAD

**Discussion by clstaudt** [fastapi/sqlmodel#1494] [Questions] [answered]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the SQLModel documentation, with the integrated search.
- [X] I already searched in Google "How to X in SQLModel" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to SQLModel but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to SQLModel but to [SQLAlchemy](https://github.com/sqlalchemy/sqlalchemy).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
class Contract(SQLModel, table=True):
    """A contract defines the business conditions of a project"""

    id: Optional[int] = Field(default=None, primary_key=True)
    title: str = Field(description="Short description of the contract.")
    client: Client = Relationship(
        back_populates="contracts",
    )
    signature_date: datetime.date = Field(
        description="Date on which the contract was signed",
    )
    start_date: datetime.date = Field(
        description="Date from which the contract is valid",
    )
    end_date: Optional[datetime.date] = Field(
        description="Date until which the contract is valid",
    )
```

### Description

I was happy to find out that every attribute of a SQLModel class can be documented with   `... = Field(description="attribute meaning here")`. Let's do more of that. However, I was confused when I tried to access and display the field description and didn't find a way to do that.

Given the example class `Contract`,   accessing `Contract.signature_date.description` gives the following output: `'signature_date'`

- This output is so counter-intuitive that it may be a bug
- How do I access the string I supplied with the `Field(description=...)` argument?
- Is there a neat way to show all the fields with their descriptions?

### Operating System

macOS

### SQLModel Version

0.0.6

### Python Version

Python 3.9.7

**Comment by byrman** [comment, marked as answer]
By accessing your model's `__fields__` attribute?
```
for k, v in Contract.__fields__.items():
    print(f"{k}: {v.field_info.description}")
```

**UPD YuriiMotov:** use `model_fields` in Pydantic V2
```py
print(Contract.model_fields["title"])
```

**Comment by clstaudt** [comment]
@byrman  Using ´Model.__fields__` works fine, thanks. This feature may be a bit hidden.  I'll leave it to others to decide whether there is a bug in here.

**Comment by StLWallace** [comment]
I've been looking for the same thing. Not sure I'd call it a bug but it would sure be cool if the Field description could be more tightly integrated with e.g. Pylance (that's probably a job for their maintainers and I'll drop a comment over there).

**Comment by starryknight64** [comment]
I had to figure this out recently, so here's what I ended up doing in case it helps anyone:
```python
from typing import Any, TypeVar

from sqlmodel import Field, SQLModel
from sqlmodel.main import FieldInfo, SQLModelMetaclass

ColumnInstrumentedAttr = TypeVar("ColumnInstrumentedAttr")


class Hero(SQLModel, table=True):
    id: int | None = Field(default=None, primary_key=True, description="The Hero's ID")
    name: str = Field(description="The Hero's name", alias="legal_name")
    secret_name: str = Field(description="Shhh! It's the Hero's secret identity!")
    age: int | None = Field(default=None, description="The Hero's age")


def colAttr(columnAttr: ColumnInstrumentedAttr, fieldName: str, defaultValue: Any = None) -> Any:
    if hasattr(columnAttr, fieldName):
        return getattr(columnAttr, fieldName)
    return defaultValue


def colAttrFromModelFields(columnAttr: ColumnInstrumentedAttr, fieldName: str, defaultValue: Any = None) -> Any:
    sqlModelMetaClass: SQLModelMetaclass = getattr(columnAttr, "class_")
    modelFields: dict[str, FieldInfo] = sqlModelMetaClass.model_fields
    fieldKey: str = getattr(columnAttr, "key")
    fieldInfo: FieldInfo = modelFields[fieldKey]
    if hasattr(fieldInfo, fieldName):
        return getattr(fieldInfo, fieldName)

    return defaultValue


def colName(columnAttr: ColumnInstrumentedAttr) -> str:
    return colAttr(columnAttr, "name")


def colDesc(columnAttr: ColumnInstrumentedAttr) -> str:
    return colAttrFromModelFields(columnAttr, "description")


print(colName(Hero.id))  # prints "id"
print(colName(Hero.name))  # prints "name"

print(colDesc(Hero.secret_name))  # prints "Shhh! It's the Hero's secret identity!"
print(colDesc(Hero.age))  # prints "The Hero's age"

print(colAttr(Hero.id, "primary_key"))  # prints "True"
print(colAttr(Hero.age, "default"))  # prints "None"

print(colAttrFromModelFields(Hero.name, "alias"))  # prints "legal_name"
print(colAttrFromModelFields(Hero.name, "something_else", "dflt"))  # prints "dflt"
```
